### PR TITLE
Check if the app user matches the news item target audience

### PR DESCRIPTION
**Pull request by abom**
If the app user does not match the target audience then lower the sort priority of the news item to 45.

Related: our-city-app/oca-backend#162

### CODE REVIEW

File: src/news/__init__.py [hunk 1]
```diff
@@ -13,4 +13,10 @@
...
 #
-# @@license_version:1.1@@
\ No newline at end of file
+# @@license_version:1.1@@
+
+# profile gender constants, rogerthat-backend UserProfile model
+GENDER_MALE_OR_FEMALE = 0
+GENDER_MALE = 1
+GENDER_FEMALE = 2
+GENDER_CUSTOM = 3
```
Comment (RubenMattan) [inline src/news/__init__.py]: Move this to a consts.py file

File: src/news/factory.py [hunk 2]
```diff
@@ -450,12 +455,63 @@ def news_updated(self, news_item):
...
+        profile (dict): user profile with birthdate and gender as long
+        news_item (dict): news item data
+
+    Returns:
+        bool
+    """
+    user_birthdate = profile.get('birthdate')
+    user_gender = profile.get('gender')
+
+    if user_birthdate is None or user_gender is None:
```
Comment (RubenMattan) [inline src/news/factory.py]: if not profile.birthdate or not profile.gender:

Gender can be 0 meaning its not set

Comment (abom) [inline src/news/factory.py]: Yes, gender can be 0 (Male or Female), but that doesn't mean it's not set, if the gender is 0 and birthdate is set, then the target audience is set.

Comment (RubenMattan) [inline src/news/factory.py]: Via the app you can only set Male or Female
Via Facebook Male, Female, Custom
And 0 means its not set

File: src/news/factory.py [hunk 3]
```diff
@@ -484,13 +484,14 @@ def match_target_audience(profile, news_item):
     if not user_birthdate or not user_gender:
         return False
 
-    news_item_gender = news_item['target_audience_gender']
+    item_target_audience = news_item['target_audience']
+    news_item_gender = item_target_audience['gender']
```
Comment (RubenMattan) [inline src/news/factory.py]: item_target_audience can be None

File: src/news/factory.py [hunk 4]
```diff
@@ -450,12 +455,67 @@ def news_updated(self, news_item):
...
+    """
+    user_birthdate = profile.get('birthdate')
+    user_gender = profile.get('gender')
+
+    if not user_birthdate or not user_gender:
+        return False
+
+    item_target_audience = news_item['target_audience']
+    if not item_target_audience:
+        return True
```
Comment (RubenMattan) [inline src/news/factory.py]: Should be placed at the beginning of the function

**Pull request merged**